File: src/main/java/org/cloudfoundry/credhub/view/JsonView.java
package org.cloudfoundry.credhub.view;

import org.cloudfoundry.credhub.credential.JsonCredentialValue;
import org.cloudfoundry.credhub.domain.JsonCredentialVersion;

@SuppressWarnings("unused")
public class JsonView extends CredentialView {

  JsonView() {  /* Jackson */ }

  JsonView(JsonCredentialVersion jsonCredential) {
    super(
        jsonCredential.getVersionCreatedAt(),
        jsonCredential.getUuid(),
        jsonCredential.getName(),
        jsonCredential.getCredentialType(),
        new JsonCredentialValue(jsonCredential.getValue())
    );
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/JsonView.kt
package org.cloudfoundry.credhub.views

import org.cloudfoundry.credhub.credential.JsonCredentialValue
import org.cloudfoundry.credhub.domain.JsonCredentialVersion

class JsonView : CredentialView {
    internal constructor() : super() /* Jackson */ {}
    internal constructor(jsonCredential: JsonCredentialVersion) : super(
        jsonCredential.versionCreatedAt,
        jsonCredential.uuid,
        jsonCredential.name,
        jsonCredential.getCredentialType(),
        JsonCredentialValue(jsonCredential.getValue()!!)
    ) {
    }
}
